### PR TITLE
osd: disable raw mode for specific disk configs

### LINKED ISSUE #9822: osdsPerDevice not working
Settings:

```
  storage:
    useAllNodes: false
    useAllDevices: false
    nodes:
      - name: "k3shost"
        devices:
          - name: "nvme0n1"
            config:
              osdsPerDevice: "2"
          - name: "nvme0n2"
            config:
              osdsPerDevice: "2"
          - name: "nvme0n3"
            config:
              osdsPerDevice: "3"
```
ceph osd tree
```
ID  CLASS  WEIGHT   TYPE NAME         STATUS  REWEIGHT  PRI-AFF
-1         0.30667  root default
-3         0.30667      host k3shost
 0   nvme  0.09769          osd.0         up   1.00000  1.00000
 1   nvme  0.10840          osd.1         up   1.00000  1.00000
 2   nvme  0.10059          osd.2         up   1.00000  1.00000
```
kubectl get all -A | grep osd
```
rook-ceph     pod/rook-ceph-osd-0-69fb4b8c9d-wjfwg                    1/1     Running     4 (43m ago)    133m
rook-ceph     pod/rook-ceph-osd-1-fcbc76895-5d7nx                     1/1     Running     2 (43m ago)    109m
rook-ceph     pod/rook-ceph-osd-2-6dc8df4949-kcq8l                    1/1     Running     0              31m
rook-ceph     pod/rook-ceph-osd-prepare-k3shost--1-8l8fd              0/1     Completed   0              31m
rook-ceph     deployment.apps/rook-ceph-osd-0                    1/1     1            1           133m
rook-ceph     deployment.apps/rook-ceph-osd-1                    1/1     1            1           109m
rook-ceph     deployment.apps/rook-ceph-osd-2                    1/1     1            1           31m
rook-ceph     replicaset.apps/rook-ceph-osd-0-69fb4b8c9d                    1         1         1       133m
rook-ceph     replicaset.apps/rook-ceph-osd-1-fcbc76895                     1         1         1       109m
rook-ceph     replicaset.apps/rook-ceph-osd-2-6dc8df4949                    1         1         1       31m
rook-ceph     job.batch/rook-ceph-osd-prepare-k3shost   1/1           15s        31m
```

**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
Only 1 osd is created per device

**Expected behavior:**
2 or 3 osds per device as configured

**Environment**:
* OS: Alpine Linux
* Kernel (e.g. `uname -a`): `Linux k3shost 5.15.16-0-lts #1-Alpine SMP Wed, 26 Jan 2022 16:46:12 +0000 x86_64 Linux`
* Cloud provider or hardware configuration: VM in VMware Workstation
* Rook version (use `rook version` inside of a Rook Pod): rook: v1.8.6 go: go1.16.14
* Storage backend version (e.g. for ceph do `ceph -v`): quay.io/ceph/ceph:v16.2.7
* Kubernetes version (use `kubectl version`): Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.6+k3s1", GitCommit:"3228d9cb9a4727d48f60de4f1ab472f7c50df904", GitTreeState:"clean", BuildDate:"2022-01-25T01:27:44Z", GoVersion:"go1.16.10", Compiler:"gc", Platform:"linux/amd64"}
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/latest/ceph-toolbox.html)): HEALTH_OK

### PR DESCRIPTION
**Pull request by BlaineEXE**
If a specific disk is configured to have more than one OSD per device,
or if it has a metadata device specified, exclude it from raw mode on an
individual basis.

Signed-off-by: Blaine Gardner <blaine.gardner@redhat.com>

**manual tests run:**
- [x] select one disk with 2 osds per device
- [x] select one disk with another disk as metadata
- [x] select one disk with 2 osds per device, another disk with default config (one should be lvm, the other raw)

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #9822 

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: pkg/daemon/ceph/osd/volume_test.go
Comment (BlaineEXE) [inline pkg/daemon/ceph/osd/volume_test.go:1410]: Same as previous test
```go
assert.Equal(t, true, isSafeToUseRawMode(sys.PartType, cephver.CephVersion{Major: 16, Minor: 2, Extra: 5}))
```

File: pkg/daemon/ceph/osd/volume_test.go
Comment (BlaineEXE) [inline pkg/daemon/ceph/osd/volume_test.go:1422]: Same as previous tests
```go
assert.Equal(t, false, isSafeToUseRawMode(sys.DiskType, cephver.CephVersion{Major: 16, Minor: 2, Extra: 5}))
assert.Equal(t, true, isSafeToUseRawMode(sys.DiskType, cephver.CephVersion{Major: 16, Minor: 2, Extra: 6}))
```

File: pkg/daemon/ceph/osd/volume.go
Comment (BlaineEXE) [inline pkg/daemon/ceph/osd/volume.go:484]: I believe this logic is equivalent to what @satoru-takeuchi had before, just written differently to reject with `false` in these if blocks and assume true at the bottom.

From what I could tell of the previous implementation, the intent was to return true (1) if it was a partition, or (2) if it was a disk AND ceph has the atari fix.

**Review by travisn — APPROVED** [review]

**Comment by BlaineEXE** [comment]
Let's wait for a review from @satoru-takeuchi as well.

**Review by satoru-takeuchi — DISMISSED** [review]
I made just one trivial comment. Please merge this PR after reflecting this.

**Pull request merged**
